resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/config.scala

```diff
@@ -225,6 +225,17 @@ package object config extends Logging {
     .timeConf(TimeUnit.MILLISECONDS)
     .createWithDefaultString("1s")
 
+  private[spark] val REPORT_LOG_FREQUENCY = {
+    ConfigBuilder("spark.yarn.report.logging.frequency")
+      .doc("Maximum number of application reports processed " +
+        "until the next application status is logged. " +
+        "If there is a change of state, the application status will be logged " +
+        "regardless of the number of application reports processed.")
+      .version("3.5.0")
+      .intConf
```

Comment:
In case of negative values, what happens? If we don't want to disable this feature, shall we add the following?
```
.checkValue(_ > 0, "value should be positive")
```

Reply:
If you want to support to an option to disable this feature, please document it additionally.